test/dx_mgn_pool.js

```diff
@@ -1,10 +1,36 @@
 const DxMgnPool = artifacts.require("DxMgnPool")
+const DutchExchange = artifacts.require("DutchExchange")
+const abi = require('ethereumjs-abi')
+
 const ERC20 = artifacts.require("ERC20")
 const MockContract = artifacts.require("MockContract")
 
 const truffleAssert = require("truffle-assertions")
 const { waitForNBlocks } = require("./utilities")
 
+const prepareParticipationMock = async( auctionIndex, sellVolume, newAuctionIndex, generatedMGN, dx, dxMock, dTMock, account) => {
```

Comment:
I'm not a fan of having these kind of methods in unit tests. I agree there is a lot of duplicated code, but tests shouldn't really follow the DRY principle but instead be read more like an individually story that documents one interaction with the unit under test (here is the setup, here are the interactions, here are the expectations). 

By having methods that do the "happy case" setup we have to jump around in the code to understand what the system has been configured with and chances are that for a specific test one value has to be different.

I find it usually easier to organize the expectation block somewhat readably (method mocked for dx, then method mocked for depositToken etc) and copy it around the tests. Again, DRY is not really a principle in testing.